site/client/blocks/ProminentLink/ProminentLink.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from "react"
+import ReactDOM from "react-dom"
+import { Grapher } from "site/client/Grapher"
+import { observer } from "mobx-react"
+import { computed } from "mobx"
+import {
+    strToQueryParams,
+    queryParamsToStr,
+    splitURLintoPathAndQueryString,
+    QueryParams
+} from "utils/client/url"
+import { EntityUrlBuilder } from "charts/ChartUrl"
+import { union, isEmpty } from "charts/Util"
+
+const PROMINENT_LINK_CLASSNAME = "wp-block-owid-prominent-link"
+
+@observer
+class ProminentLink extends React.Component<{
+    originalURL: string
+    innerHTML: string | null
+}> {
+    @computed get originalURLPath() {
+        return splitURLintoPathAndQueryString(this.props.originalURL).path
+    }
+
+    @computed get originalURLQueryString(): string | undefined {
+        return splitURLintoPathAndQueryString(this.props.originalURL)
+            .queryString
+    }
+
+    @computed get originalURLQueryParams(): QueryParams | undefined {
+        const { originalURLQueryString } = this
```

Comment:
nit: if `originalURLQueryString` is only used here, maybe ditch the method and do it inline?

Reply:
So I'm curious about your philosophy here. I sometimes prefer to separate the name from the access logic to maximize readability and keep each function simple. The downside is a bit of indirection as noted here: https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

With modern code editors/instant previews of function definitions, I find most of the cost can basically be avoided, and that the primary benefits actually accrue to the reader (they can understand the point of the code rather than the access logic). What do you think?

Reply:
Either way is fine. I was thinking if it's only used once, maybe `const originalURLQueryString = splitURLintoPathAndQueryString(this.props.originalURL).queryString` is slightly better because fewer tokens and smaller API surface (the latter can also be achieved by making the method private). But super nit and I think either way is fine and your implementation is better if we were to need that logic twice.